accumulo-store/src/main/java/gaffer/accumulostore/retriever/impl/AccumuloIDBetweenSetsRetriever.java

```diff
@@ -131,15 +141,40 @@ protected boolean checkIfBothEndsInSet(final Object source, final Object destina
 
     private class ElementIteratorFromBatches extends AbstractElementIteratorFromBatches {
         ElementIteratorFromBatches() throws RetrieverException {
-            addToBloomFilter(seedSetB, filter, clientSideFilter);
-            addToBloomFilter(seedSetA, filter, clientSideFilter);
-            idsAIterator = seedSetA.iterator();
+            addToBloomFilter(seedSetBIter, filter, clientSideFilter);
+            idsAIterator = seedSetAIter;
             updateScanner();
         }
 
         @Override
         protected void updateBloomFilterIfRequired(final EntitySeed seed) throws RetrieverException {
             // no action required.
         }
+
+        protected boolean secondaryCheck(final Element elm) {
+            if (Entity.class.isInstance(elm)) {
+                return true;
+            }
+            final Edge edge = (Edge) elm;
+            final Object source = edge.getSource();
+            final Object destination = edge.getDestination();
+            final boolean sourceIsInCurrent = currentSeeds.contains(source);
+            final boolean destIsInCurrent = currentSeeds.contains(destination);
+            boolean sourceMatchesClientFilter;
+            try {
+                sourceMatchesClientFilter = clientSideFilter.membershipTest(
+                        new org.apache.hadoop.util.bloom.Key(elementConverter.serialiseVertexForBloomKey(source)));
+            } catch (final AccumuloElementConversionException e) {
+                return false;
+            }
+            boolean destMatchesClientFilter;
+            try {
+                destMatchesClientFilter = clientSideFilter.membershipTest(
+                        new org.apache.hadoop.util.bloom.Key(elementConverter.serialiseVertexForBloomKey(destination)));
+            } catch (final AccumuloElementConversionException e) {
+                return false;
+            }
+            return (sourceIsInCurrent && destMatchesClientFilter) || (destIsInCurrent && sourceMatchesClientFilter);
```

Comment:
This could be made slightly more efficient if we do (sourceIsInCurrent && destMatchesClientFilter) check and if true we could skip the calculation for destIsInCurrent and sourceMatchesClientFilter.